bool check_obs

---

bool obs_detect